### ISSUE TITLE
"Failed to update rosdep, did you run 'rosdep init' first?" via bloom

### ISSUE THREAD

**Issue by 130s** [ros-infrastructure/bloom#386]
Log output looks similar to https://github.com/ros-infrastructure/rosdep/issues/306 but I open a separate ticket on `bloom` since `rosdep update` by itself seems to have finished as expected as cited.

I'm releasing against `Kinetic` Should I run bloom from Kinetic-compatible distro?

```
$ lsb_release -c
Codename:	trusty

$ rosdep update
reading in sources list data from /etc/ros/rosdep/sources.list.d
Hit https://raw.githubusercontent.com/ros/rosdistro/master/rosdep/osx-homebrew.yaml
Hit https://raw.githubusercontent.com/ros/rosdistro/master/rosdep/base.yaml
Hit https://raw.githubusercontent.com/ros/rosdistro/master/rosdep/python.yaml
Hit https://raw.githubusercontent.com/ros/rosdistro/master/rosdep/ruby.yaml
Hit https://raw.githubusercontent.com/ros/rosdistro/master/releases/fuerte.yaml
Query rosdistro index https://raw.githubusercontent.com/ros/rosdistro/master/index.yaml
Add distro "groovy"
Add distro "hydro"
Add distro "indigo"
Add distro "jade"
Add distro "kinetic"
updated cache in /home/flatnoodle/.ros/rosdep/sources.cache

bloom-release --rosdistro kinetic --track kinetic ar_track_alvar_msgs
:
Expanding 'debian/rules.em' -> 'debian/rules'
Expanding 'debian/control.em' -> 'debian/control'
Creating tag: debian/ros-kinetic-ar-track-alvar-msgs_0.5.1-0_jessie
####
#### Successfully generated 'jessie' debian for package 'ar_track_alvar_msgs' at version '0.5.1-0'
####


==> git-bloom-generate -y rosrpm --prefix release/kinetic kinetic -i 0
Generating source RPMs for the packages: ['ar_track_alvar_msgs']
RPM Incremental Version: 0
RPM Distributions: ['23', '24']
Releasing for rosdistro: kinetic

Pre-verifying RPM dependency keys...
Running 'rosdep update'...
Traceback (most recent call last):
  File "/usr/lib/python2.7/dist-packages/bloom/generators/common.py", line 92, in update_rosdep
    rosdep2.catkin_support.update_rosdep()
  File "/usr/lib/python2.7/dist-packages/rosdep2/catkin_support.py", line 91, in update_rosdep
    call(('rosdep', 'update'), pipe=PIPE)
  File "/usr/lib/python2.7/dist-packages/rosdep2/catkin_support.py", line 48, in call
    raise CalledProcessError(retcode, command)
CalledProcessError: Command '('rosdep', 'update')' returned non-zero exit status 1

Failed to update rosdep, did you run 'rosdep init' first?
<== Error running command '['/usr/bin/git-bloom-generate', '-y', 'rosrpm', '--prefix', 'release/kinetic', 'kinetic', '-i', '0']'
Release failed, exiting.

$ dpkg -p python-bloom python-rosdep|grep Ver
Version: 0.5.21-1
Version: 0.11.4-1
```

**Comment by wjwwood** [comment]
I can't reproduce this. Does it fail every time? What platform you run bloom on doesn't matter (I use OS X).

**Comment by wjwwood** [comment]
Also, the key point from https://github.com/ros-infrastructure/rosdep/issues/306 is that he was running bloom in parallel. So I think the theory is that `rosdep update` is not safe to run concurrently. So don't run bloom concurrently either and you should be ok I would think.

Otherwise, `rosdep update` may be failing for an entirely different reason.

**Comment by mikeferguson** [comment]
We're seeing this on our buildfarms -- but it looks like it is a github issue (we're seeing 503 responses mixed in)

**Comment by wjwwood** [comment]
I just saw some `connection reset by peer`'s as well (but in a different part of bloom):

```
==> Looking for a release of this repository in a different distribution...
Traceback (most recent call last):
  File "/usr/local/bin/bloom-release", line 9, in <module>
    load_entry_point('bloom==0.5.21', 'console_scripts', 'bloom-release')()
  File "/usr/local/lib/python2.7/site-packages/bloom-0.5.21-py2.7.egg/bloom/commands/release.py", line 1322, in main
    args.override_release_repository_push_url)
  File "/usr/local/lib/python2.7/site-packages/bloom-0.5.21-py2.7.egg/bloom/commands/release.py", line 1168, in perform_release
    release_repo = get_release_repo(repository, distro, override_release_repository_url)
  File "/usr/local/lib/python2.7/site-packages/bloom-0.5.21-py2.7.egg/bloom/commands/release.py", line 358, in get_release_repo
    url = get_repo_uri(repository, distro)
  File "/usr/local/lib/python2.7/site-packages/bloom-0.5.21-py2.7.egg/bloom/commands/release.py", line 325, in get_repo_uri
    default_distro, default_release = get_most_recent('release', repository)
  File "/usr/local/lib/python2.7/site-packages/bloom-0.5.21-py2.7.egg/bloom/commands/release.py", line 240, in get_most_recent
    distro_file = get_distribution_file(distro)
  File "/usr/local/lib/python2.7/site-packages/bloom-0.5.21-py2.7.egg/bloom/commands/release.py", line 255, in get_distribution_file
    files = rosdistro.get_distribution_files(get_index(), distro)
  File "/usr/local/lib/python2.7/site-packages/rosdistro/__init__.py", line 138, in get_distribution_files
    data = _get_dist_file_data(index, dist_name, 'distribution')
  File "/usr/local/lib/python2.7/site-packages/rosdistro/__init__.py", line 324, in _get_dist_file_data
    data.append(_load_yaml_data(u))
  File "/usr/local/lib/python2.7/site-packages/rosdistro/__init__.py", line 316, in _load_yaml_data
    yaml_str = load_url(url)
  File "/usr/local/lib/python2.7/site-packages/rosdistro/loader.py", line 59, in load_url
    raise URLError(str(e) + ' (%s)' % url)
URLError: <urlopen error <urlopen error [Errno 54] Connection reset by peer> (https://raw.githubusercontent.com/ros/rosdistro/994bafe52ae53bb1e33686597bda2b35ac09a6b6/hydro/distribution.yaml)>
```

**Comment by 130s** [comment]
Ok, after 7 attempts it went through. I also saw `URLError` (with groovy/distribution.yaml) once. Maybe just a network glitch on github or somewhere.
